MsGraphicsPkg/Library/BaseUiRectangleLib/UiRectangle.c

```diff
@@ -59,14 +64,53 @@ IN UI_STYLE_INFO* StyleInfo
     return NULL;
   }
 
-  FillDataSize = GetFillDataSize(Width, Height, StyleInfo);
+  DEBUG((DEBUG_INFO, "MATTHEW CARLSON START\n"));
+
+  // This is the default arrangement - 32 bit RGB
+  if (PixelFormat == PixelRedGreenBlueReserved8BitPerColor) {
+    PixelSizeInBytes = 4;
+  }
+
+  // This is the a reverse arrangement of the standard 32 bit BGR (RGB reversed)
+  if (PixelFormat == PixelBlueGreenRedReserved8BitPerColor) {
+    // TODO: Figure out how to re-organize the colors and create a color mapping
+    DEBUG((DEBUG_ERROR, "%a:%d - BGR revered Framebuffer BOPS are not supported currently\n", __FILE__,__LINE__));
+    ASSERT(PixelFormat != PixelBlueGreenRedReserved8BitPerColor);
+
+  }
+  // This is currently unsupported by this library
+  else if (PixelFormat == PixelBltOnly) {
+    DEBUG((DEBUG_ERROR, "%a:%d - BLT only GOP's are not supported currently\n", __FILE__,__LINE__));
+    ASSERT(PixelFormat != PixelBltOnly);
+  }
+  // If this is bitmask, this mean to check the PixelBitMap and figure out how to handle it
+  else if (PixelFormat == PixelBitMask) {
+    DEBUG((DEBUG_INFO, "Figuring out how to map values from bitmap onto pixel size\n"));
+    ASSERT(PixelFormatBitMap != NULL);
+    UINT32 PixelBitFormat = PixelFormatBitMap->BlueMask | PixelFormatBitMap->GreenMask | PixelFormatBitMap->RedMask | PixelFormatBitMap->ReservedMask;
+    ASSERT(PixelBitFormat != 0); // make sure we have a valid format
+    DEBUG((DEBUG_INFO, "GOP Pixel Format: %x\n", PixelBitFormat));
+    UINT8  BitsInPixel = 1;
+    while( PixelBitFormat>>=1 ) BitsInPixel++;
+    DEBUG((DEBUG_INFO, "GOP Bits in Pixel: %d\n", BitsInPixel));
+    PixelSizeInBytes = BitsInPixel / 8;
```

Comment:
what happens if BitsInPexel equals a size not power of 2?  You don't check that here and would cause you to corrupt the buffer.

Reply:
If the bits per pixel is not a power of two, then we have bigger problems. I'm open to solutions though. We should probably round up.